gst/nnstreamer/tensor_common.c

```diff
@@ -1225,8 +1224,40 @@ gst_gen_tensors_from_collectpad (GstCollectPads * collect,
       }
         break;
       case SYNC_NOSYNC:
-        gst_buffer_unref (buf);
-        buf = gst_collect_pads_pop (collect, data);
+        if (buf != NULL) {
+          gst_buffer_unref (buf);
+          buf = gst_collect_pads_pop (collect, data);
+        }
+        break;
+      case SYNC_BASEPAD:
+      {
```

Comment:
This brace looks strange. Is it neccessary?

Reply:
done.